### PR TITLE
Show an error dialog if we fail to send location

### PR DESCRIPTION
**Pull request by andybalaam**
image

---
This PR currently has no changelog labels, so will not be included in changelogs.

A reviewer can add one of: `T-Deprecation`, `T-Enhancement`, `T-Defect`, `T-Task` to indicate what type of change this is, or add `Type: [enhancement/defect/task]` to the description and I'll add them for you.

Preview: https://61e026eae955de51de6e9dbb--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/views/location/LocationButton.tsx [hunk 1]
```diff
@@ -75,6 +81,38 @@ export const LocationButton: React.FC<IProps> = (
...
+                    onFinished: (tryAgain: boolean) => {
+                        if (tryAgain) {
+                            openMenu();
+                        }
+                    },
+                },
+            );
+        }
+        return true;
+    };
```
Comment (t3chguy) [inline src/components/views/location/LocationButton.tsx:114]: floating braces are a sign of poor indentation elsewhere imo

Comment (andybalaam) [reply src/components/views/location/LocationButton.tsx:114]: How would you improve this section?  Nothing jumps out at me.

File: src/components/views/location/LocationButton.tsx [hunk 2]
```diff
@@ -17,21 +17,27 @@ limitations under the License.
...
 
 import { _t } from '../../../languageHandler';
 import LocationPicker from './LocationPicker';
 import { CollapsibleButton, ICollapsibleButtonProps } from '../rooms/CollapsibleButton';
 import ContextMenu, { aboveLeftOf, useContextMenu, AboveLeftOf } from "../../structures/ContextMenu";
+import Modal from '../../../Modal';
+import QuestionDialog from '../dialogs/QuestionDialog';
 
 interface IProps extends Pick<ICollapsibleButtonProps, "narrowMode"> {
+    client: MatrixClient;
```
[the remark this answers is not in the document]
Comment (andybalaam) [reply src/components/views/location/LocationButton.tsx]: Thank you, fixed in 2e437dafc1

File: src/components/views/location/LocationButton.tsx [hunk 3]
```diff
@@ -75,6 +82,38 @@ export const LocationButton: React.FC<IProps> = (
...
+                    onFinished: (tryAgain: boolean) => {
+                        if (tryAgain) {
+                            openMenu();
+                        }
+                    },
+                },
+            );
+        }
+        return true;
+    };
```
Comment (t3chguy) [inline src/components/views/location/LocationButton.tsx:115]:
```suggestion
const shareLocation = (
    client: MatrixClient,
    roomId: string,
    openMenu: () => void,
) => (
    uri: string,
    ts: number,
): boolean => {
    if (!uri) return false;
    try {
        const text = textForLocation(uri, ts, null);
        client.sendMessage(
            roomId,
            makeLocationContent(text, uri, ts, null),
        );
    } catch (e) {
        logger.error("We couldn’t send your location", e);
        Modal.createTrackedDialog(
            'We couldn’t send your location',
            '',
            QuestionDialog,
            {
                title: _t("We couldn’t send your location"),
                description: _t(
                    "Element could not send your location. Please try again later."),
                button: _t('Try again'),
                cancelButton: _t('Cancel'),
                onFinished: (tryAgain: boolean) => {
                    if (tryAgain) {
                        openMenu();
                    }
                },
            },
        );
    }
    return true;
};
```

perhaps something like this, I don't think its massively more readable so maybe not using the shorthand inline return on the arrow func here and using braces and an actual return might be cleaner

**Review by t3chguy — APPROVED** [review]
Looks sane other than styling nit

File: src/components/views/location/LocationButton.tsx (hunk 3 above)
Comment (andybalaam) [reply src/components/views/location/LocationButton.tsx]: I attempted to make it better by breaking the args into separate variables in bdcf002 .

**Pull request merged**